indico/htdocs/js/indico/modules/sessions.js

```diff
@@ -0,0 +1,130 @@
+(function(global) {
+    'use strict';
+
+    function handleRowSelecteion() {
+        $('table.i-table input.select-row').on('change', function() {
+            $(this).closest('tr').toggleClass('selected', this.checked);
+            $('.js-requires-selected-row').toggleClass('disabled', !$('.sessions input:checkbox:checked').length)
+        }).trigger('change');
+    }
+
+    function setupTableSorter() {
+        $('.sessions .tablesorter').tablesorter({
+            cssAsc: 'header-sort-asc',
+            cssDesc: 'header-sort-desc',
+            cssInfoBlock: 'avoid-sort',
+            cssChildRow: 'session-blocks-row',
+            headerTemplate: '',
+            sortList: [[1, 0]]
+        });
+    }
+
+    global.setupSessionsList = function setupSessionsList() {
+        handleRowSelecteion();
+        setupTableSorter();
+
+        $('#select-all').on('click', function() {
+            $('table.i-table input.select-row').prop('checked', true).trigger('change');
+        });
+
+        $('#select-none').on('click', function() {
+            $('table.i-table input.select-row').prop('checked', false).trigger('change');
+        });
+
+        $('.sessions .show-session-slots').on('click', function() {
+            $(this).parent().parent().next('tbody.session-blocks').toggle();
+        });
+
+        $('.sessions .toolbar').on('click', '.disabled', function(e) {
+            e.preventDefault();
+            e.stopPropagation();
+        });
+
+        $('#add-new-session').on('click', function() {
+            var $this = $(this);
+            ajaxDialog({
+                url: $this.data('href'),
+                title: $this.data('title'),
+                onClose: function(data) {
+                    if (data) {
+                        window.location.reload();
```

Comment:
Better re-render the rows and update the document.

You'll have to move the code that renders the rows into a jinja macro (so you can call it on a template module - check existing code where we do it) and to convert some event handlers to use delegate syntax so you don't have to re-register them but besides that it's pretty simply.